### ISSUE TITLE
Grid layout looks nasty

### ISSUE THREAD

**Issue by lawsie** [lawsie/guizero#18]
Grid layout looks nasty - things are all squashed up and there is no padding. Suggestions for good and easy ways to improve grid layouts? Here are some ideas I had:

- Making a "blank" widget which just creates a blank space of a specified width/height?
- Making a "<hr>" type widget which would just create a horizontal line (spanning multiple cols?)
- Just use existing methods in tkinter for changing col/row dimensions, but document them (e.g. http://infohost.nmt.edu/tcc/help/pubs/tkinter/web/grid-config.html) ?
- Encourage the use of lots of Box widgets to divide up the gui 
- Allow rowspans and colspans to be specified for widgets (already exists in tkinter)

**Comment by martinohanlon** [comment]
I've just used guizero for some prototyping and a rowspan / colspan would be really useful.

**Comment by penguintutor** [comment]
I agree that columspan & rowspan would be useful.

I've had a look at how this is called in utilities.py and it looks like it's a fairly easy thing to add.

I'm happy to have a go at adding that feature myself, but how best to implement? For example this could be supported by adding a columnspan and rowspan as an option in each of the widgets (which presumably would need an update in each of the widgets as well as utilities.py), but an easier way (and arguably more logical) would be to have it within the list where the grid position is passed.

So you could put: grid=[x,y] (as currently without any columnspan / rowspan), or grid=[x,y,xspan,yspan], where xspan provides columnspan value and yspan is the rowspan (where the grid list must be either 2 or 4 entries).

Does that sound like it would be worth adding?

**Comment by martinohanlon** [comment]
I would tend to favour a  grid=[x,y,xspan,yspan] approach, that way you could have the same interface for either widgets that span and widgets that dont.

Will tkinter support breaking the grid layout? Allowing you to span?

I have been considering implementing a new 'layout' which is neither pack or grid? A more flexible layout which allows more dynamic layout, made up of individually placed boxes which widgets can be placed into. (I just havent got past the considering phase.

**Comment by penguintutor** [comment]
I do like the idea of a new layout style that makes it easier to use. Layouts are one of the tricky things for those new to GUI programming (including kids). In the short term I think it was worth looking at what can be used with the existing grid layout to make it more useful.

I also prefer the grid=[x,y,xspan,yspan] approach. The columnspan and rowspan are implemented as a feature of tkinter and so adding it within the utilities means it should work for any widget in guizero with no other changes necessary to the widgets themselves.

The following screenshot shows how this can be used with the PushButton keyboard example included in the documentation. Button 1 has a columnspan and button 2 has a rowspan value.  I did need to add additional padding to the buttons to make it look this way, but that will not be required if using appropriate sized widgets (eg. images).

![image](https://user-images.githubusercontent.com/5587863/34386500-db476ea6-eb20-11e7-8901-5200da0ff1ef.png)

I've created my own branch and made the necessary changes. It's only an extra 13 lines of code in utilities.py 

I will need to update the documents before this is added as a feature. I think an image example may work better than the keyboard example I've shown above. Feel free to comment on what you think about what I've done so far, if there are no objections I'll look at adding this into the documentation and issuing a pull-request.


The code for the screenshot example is below (cut down version of the keyboard example)
```
from guizero import App, PushButton
def do_nothing():
    print("Nothing happened")
app = App(title="Keypad example", width=120, height=100, layout="grid")
button1 = PushButton(app, command=do_nothing, text="1", grid=[0,0,2,1], padx=26)
button2 = PushButton(app, command=do_nothing, text="2", grid=[2,0,1,2], pady=30)
button4  = PushButton(app, command=do_nothing, text="4", grid=[0,1])
button5  = PushButton(app, command=do_nothing, text="5", grid=[1,1])
app.display()
```
